code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm

```diff
@@ -0,0 +1,222 @@
+
+// --------------------------------------------------------------------------------
+// ----------------- TERROR SPIDERS: T4 QUEEN OF TERROR ---------------------------
+// --------------------------------------------------------------------------------
+// -------------: ROLE: gamma-level threat to the whole station, like a blob
+// -------------: SPECIAL: spins webs, breaks lights, breaks cameras, webs objects, lays eggs, commands other spiders...
+// -------------: TO FIGHT IT: bring an army, and take no prisoners. Mechs and/or decloner guns are a very good idea.
+// -------------: SPRITES FROM: FoS, http://nanotrasen.se/phpBB3/memberlist.php?mode=viewprofile&u=386
+
+/mob/living/simple_animal/hostile/poison/terror_spider/queen
+	name = "Queen of Terror spider"
+	desc = "An enormous, terrifying spider. Its egg sac is almost as big as its body, and teeming with spider eggs!"
+	spider_role_summary = "Commander of the spider forces. Lays eggs, directs the brood."
+
+	icon_state = "terror_queen"
+	icon_living = "terror_queen"
+	icon_dead = "terror_queen_dead"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	move_to_delay = 15 // yeah, this is very slow, but
+	ventcrawler = 1
+	var/spider_spawnfrequency = 1200 // 120 seconds
+	var/spider_spawnfrequency_stable = 1200 // 120 seconds. Spawnfrequency is set to this on ai spiders once nest setup is complete.
+	var/spider_lastspawn = 0
+	var/nestfrequency = 150 // 15 seconds
+	var/lastnestsetup = 0
+	var/neststep = 0
+	var/hasnested = 0
+	var/spider_max_per_nest = 20 // above this, queen stops spawning more, and declares war.
+
+	var/canlay = 0 // main counter for egg-laying ability! # = num uses, incremented at intervals
+	var/spider_can_fakelings = 3 // spawns defective spiderlings that don't grow up, used to freak out crew, atmosphere
+
+	force_threshold = 18 // outright immune to anything of force under 18, this means welders can't hurt it, only guns can
+
+	ranged = 1
+	retreat_distance = 5
+	minimum_distance = 5
+	projectilesound = 'sound/weapons/pierce.ogg'
+	projectiletype = /obj/item/projectile/terrorqueenspit
+
+	spider_tier = TS_TIER_4
+	spider_opens_doors = 2
+	loot = list(/obj/item/clothing/accessory/medal)
+
+	var/datum/action/innate/terrorspider/queen/queennest/queennest_action
+	var/datum/action/innate/terrorspider/queen/queensense/queensense_action
+	var/datum/action/innate/terrorspider/queen/queeneggs/queeneggs_action
+	var/datum/action/innate/terrorspider/queen/queenfakelings/queenfakelings_action
+
+/mob/living/simple_animal/hostile/poison/terror_spider/queen/New()
+	..()
+	queennest_action = new()
+	queennest_action.Grant(src)
+	spider_myqueen = src
+	if(spider_awaymission)
+		spider_growinstantly = 1
+		spider_spawnfrequency = 150
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/queen/Life()
+	..()
+	if(stat != DEAD)
```

Comment:
You should replace this with
```
. = ..()
if(.)
```
/mob/living/Life() returns true if `stat != DEAD`.